addons/account_ubl/tests/test_ubl.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from odoo import tools
+from odoo.tests import common
+
+import logging
+import base64
+
+from lxml import etree
+
+_logger = logging.getLogger(__name__)
+
+UBL_NAMESPACES = {
+    'cac': '{urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2}',
+    'cbc': '{urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2}',
+}
+
+class TestUBL(common.TransactionCase):
+    def test_ubl_invoice(self):
+        Invoice = self.env['account.invoice']
+        invoices_ids = Invoice.search([])
+        for invoice in invoices_ids:
```

Comment:
Please this tests brake all rules for unit tests:

- It has for's on it.
- A lot of IF.
- Tries.

IMHO, we need some kind of meta framework for tests (depending of things like country) add several nested IF in tests is not correct.